app1.js

```diff
@@ -0,0 +1,115 @@
+const userDay = document.querySelector("#userDay");
+const userMonth = document.querySelector("#userMonth");
+const userYear = document.querySelector("#userYear");
+const spanDay = document.querySelector("#spanDay");
+const spanMonth = document.querySelector("#spanMonth");
+const spanYear = document.querySelector("#spanYear");
+const button = document.querySelector("button");
+const dayErr = document.querySelector("#dayErr");
+const monthErr = document.querySelector("#monthErr");
+const yearErr = document.querySelector("#yearErr");
+const labelDay = document.querySelector("#labelDay");
+const labelMonth = document.querySelector("#labelMonth");
+const labelYear = document.querySelector("#labelYear");
+
+const currentDate = new Date();
+
+const currentYear = currentDate.getFullYear();
+const currentMonth = currentDate.getMonth();
+const currentDay = currentDate.getDate();
+const fullDate = new Date(currentYear, currentMonth, currentDay, 0, 0, 0, 0);
+console.log(fullDate);
+
+function calculateDifference() {
+  let userDate = new Date(
+    userYear.value,
+    userMonth.value - 1,
+    userDay.value,
+    0,
+    0,
+    0,
+    0,
+  );
+  let differenceMs = Math.abs(fullDate - userDate);
+  const oneDay = 1000 * 60 * 60 * 24;
+  const oneMonth = oneDay * 30.44;
+  const oneYear = oneDay * 365.25;
+  const years = Math.floor(differenceMs / oneYear);
+  const months = Math.floor((differenceMs % oneYear) / oneMonth);
+  const days = Math.floor((differenceMs % oneMonth) / oneDay);
+  return { years, months, days };
+}
+
+function calculatedResults(years, months, days) {
+  spanDay.innerText = days;
+  spanMonth.innerText = months;
+  spanYear.innerText = years;
+}
+
+function preventResults() {
+  spanDay.innerText = "--";
+  spanMonth.innerText = "--";
+  spanYear.innerText = "--";
+}
+
+function showError(input, errorElement, labelColor, errorMessage) {
+  preventResults();
+  errorElement.innerText = errorMessage;
+  errorElement.style.display = "block";
+  input.style.borderColor = "red";
+  labelColor.style.color = "red";
+}
+
+function resetError(input, errorElement, labelColor) {
+  errorElement.style.display = "none";
+  input.style.borderColor = "";
+  labelColor.style.color = "hsl(0, 1%, 44%)";
```

Comment:
Ovo se moze positici s CSS-om. U JS-u nam zivi logika, i jos ako bude se trebao baviti bojama i stilovima postaje jako kompleksan i tezak za citati. Uvijek gledamo da u JS-u imamo sto manje logike koja se bavi prezentacijom, ukoliko je to moguce.

```html
<div id="month-wrapper" class="input-wrapper">
  <label>Month</label>
  <input type="text" placeholder="MM" id="userMonth" />
  <p class="error-message">This field is required</p>
</div>
```

```js
const monthWrapper = document.getElementById('month-wrapper');

// ....

function addError(wrapperElement, errorMessage) {
  wrapperElement.classList.add('error');
  wrapperElement.getElementByClassName('error-message')[0]?.innerText = errorMessage;
}

function removeError(wrapperElement) {
  wrapperElement.classList.remove('error');
  wrapperElement.getElementByClassName('error-message')[0]?.innerText = '';
}

// ...

addError(monthWrapper, 'Some error');
```

```css
.input-wrapper {
  .error-message {
    display: none;
  }
}

.input-wrapper.error {
  label {
    color: red;
  }

  input {
    border-color: red;
  }

  .message {
    display: block;
    color: red;
  }
}

```

Ovako smo razdvojili prezentacijsku "logiku", te prilikom izmjena izgleda diramo samo HTML i CSS a ne JS.